queue/src/fiskaltrust.Middleware.Localization.QueueES/Processors/ReceiptCommandProcessorES.cs

```diff
@@ -6,45 +6,108 @@
+            }
+
+            var referencedQueueItems = await queueItemRepository.GetByReceiptReferenceAsync(request.ReceiptRequest.cbPreviousReceiptReference.SingleValue).ToListAsync();
+            if (!referencedQueueItems.Any())
+            {
+                throw new Exception($"Referenced queue item with reference {request.ReceiptRequest.cbPreviousReceiptReference.SingleValue} not found.");
+            }
+            if (referencedQueueItems.Count > 1)
+            {
+                throw new Exception($"Multiple queue items found with reference {request.ReceiptRequest.cbPreviousReceiptReference.SingleValue}.");
+            }
+            referencedQueueItem = referencedQueueItems.Single();
+        }
+
+        if (request.ReceiptResponse.ftStateData is not null)
+        {
+            throw new Exception("ftStateData must be empty.");
+        }
+
+        request.ReceiptResponse.ftStateData = new MiddlewareState
+        {
+            ES = new MiddlewareQueueESState
+            {
+                LastReceipt = lastQueueItem is null ? null : new LastReceipt
+                {
+                    Request = JsonSerializer.Deserialize<ReceiptRequest>(lastQueueItem.request)!,
+                    Response = JsonSerializer.Deserialize<ReceiptResponse>(lastQueueItem.response)!,
+                }
+            }
+        };
+
         var response = await (await _essscd).ProcessReceiptAsync(new ProcessRequest
         {
             ReceiptRequest = request.ReceiptRequest,
             ReceiptResponse = request.ReceiptResponse,
-            PreviousReceiptRequest = previousQueueItem is null ? null : JsonSerializer.Deserialize<ReceiptRequest>(previousQueueItem.request)!, // handle null case?
-            PreviousReceiptResponse = previousQueueItem is null ? null : JsonSerializer.Deserialize<ReceiptResponse>(previousQueueItem.response)!,
+            ReferencedReceiptRequest = referencedQueueItem is null ? null : JsonSerializer.Deserialize<ReceiptRequest>(referencedQueueItem.request)!,
+            ReferencedReceiptResponse = referencedQueueItem is null ? null : JsonSerializer.Deserialize<ReceiptResponse>(referencedQueueItem.response)!,
         });
 
+        try
+        {
+            var responseStateData = JsonSerializer.Deserialize<MiddlewareState>(((JsonElement)response.ReceiptResponse.ftStateData!).GetRawText())!;
+            await (await _journalESRepository).InsertAsync(new ftJournalES
+            {
+                ftJournalESId = Guid.NewGuid(),
+                Number = response.ReceiptResponse.ftQueueRow,
+                Data = JsonSerializer.Serialize(responseStateData.ES!.GovernmentAPI),
+                JournalType = JournalESType.VeriFactu.ToString(),
+                ftQueueItemId = response.ReceiptResponse.ftQueueItemID,
+                ftQueueId = response.ReceiptResponse.ftQueueID,
+            });
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "Error processing receipt");
```

Comment:
We could first save the queueitem into the queue below and then throw if this fails but the receipt has been sent to verifactu in this case so I don't think that would be correct... should we add an error message as a signature?